src/libData/AccountData/AccountStoreSC.tpp

```diff
@@ -159,21 +159,28 @@ bool AccountStoreSC<MAP>::UpdateAccounts(
     }
 
     bool init = true;
-    // Initiate the contract account, including setting the contract code
-    // store the immutable states
-    if (!toAccount->InitContract(transaction.GetCode(), transaction.GetData(),
-                                 toAddr, blockNum, temp)) {
-      LOG_GENERAL(WARNING, "InitContract failed");
-      init = false;
-    }
 
-    m_curBlockNum = blockNum;
-    if (init && !ExportCreateContractFiles(*toAccount)) {
-      LOG_GENERAL(WARNING, "ExportCreateContractFiles failed");
-      init = false;
-    }
+    try {
+      // Initiate the contract account, including setting the contract code
+      // store the immutable states
+      if (!toAccount->InitContract(transaction.GetCode(), transaction.GetData(),
+                                   toAddr, blockNum, temp)) {
+        LOG_GENERAL(WARNING, "InitContract failed");
+        init = false;
+      }
+
+      m_curBlockNum = blockNum;
+      if (init && !ExportCreateContractFiles(*toAccount)) {
+        LOG_GENERAL(WARNING, "ExportCreateContractFiles failed");
+        init = false;
+      }
 
-    if (init && !this->DecreaseBalance(fromAddr, gasDeposit)) {
+      if (init && !this->DecreaseBalance(fromAddr, gasDeposit)) {
+        init = false;
+      }
+    } catch (const std::exception& e) {
+      LOG_GENERAL(WARNING,
```

Comment:
It might be beneficial to add more logging as to what inpur caused the exception

Reply:
Yeah agree, but it needs more refined exception handling structure. The current one is more likely to serve for avoid crashes so its scope is a bit wide, which makes it difficult to log down everything inside the `try` field.